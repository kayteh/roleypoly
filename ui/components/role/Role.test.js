/* eslint-env jest */

import * as React from 'react'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'
import Role from './index'

describe('<Role />', () => {
  it('renders correctly', () => {
    const role = renderer.create(<Role role={{ name: 'Test Role', color: '#ffffff' }} />)
    expect(role).toMatchSnapshot()
  })

  it('triggers onToggle with new state', () => {
    let changed = false
    const role = shallow(
      <Role
        role={{ name: 'Test Role', color: '#ffffff' }}
        onToggle={(next) => { changed = next }}
        active={false}
      />
    )
    role.simulate('click')
    expect(changed).toBe(true)

    const role2 = shallow(
      <Role
        role={{ name: 'Test Role', color: '#ffffff' }}
        onToggle={(next) => { changed = next }}
        active
      />
    )
    role2.simulate('click')
    expect(changed).toBe(false)
  })

  it('fixes colors when they are not set', () => {
    const role = shallow(<Role role={{ name: 'Test Role', color: 0 }} />)
    expect(role.props().style['--role-color-base']).toEqual('hsl(0, 0%, 93.7%)')
  })

  it('has a single space for a name when empty', () => {
    const role = shallow(<Role role={{ name: '', color: '#ffffff' }} />)
    expect(role.text()).toEqual(' ')
  })

  describe('when disabled,', () => {
    it('handles touch hover events', () => {
      const role = shallow(<Role role={{ name: 'unsafe role', color: '#ffffff' }} disabled />)

      role.simulate('touchstart')
      expect(role.state().hovering).toEqual(true)
      expect(role.html()).toMatchSnapshot() // expecting tooltip
      expect(role.exists('tooltip')).toEqual(true)

      role.simulate('touchend')
      expect(role.state().hovering).toEqual(false)
    })

    it('does not trigger onToggle on click', () => {
      let changed = false
      const role = shallow(
        <Role
          role={{ name: 'Test Role', color: '#ffffff' }}
          onToggle={() => { changed = true }}
          active={changed}
          disabled
        />
      )
      expect(role.html()).toMatchSnapshot()
      role.simulate('click')

      expect(role.html()).toBe(role.html())
      expect(changed).toBe(false)
    })
  })
})
