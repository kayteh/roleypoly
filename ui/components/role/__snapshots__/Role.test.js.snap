// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Role /> renders correctly 1`] = `
<div
  className="rolestyled-r4hjov-0 kfzQxT"
  onClick={[Function]}
  onTouchEnd={null}
  onTouchStart={null}
  style={
    Object {
      "--role-color-active": "hsl(0, 0%, 100%)",
      "--role-color-base": "hsl(0, 0%, 100%)",
      "--role-color-outline": "hsla(0, 0%, 100%, 0.7)",
      "--role-color-outline-alt": "hsla(0, 0%, 100%, 0.4)",
    }
  }
  title={null}
>
  Test Role
</div>
`;

exports[`<Role /> when disabled, does not trigger onToggle on click 1`] = `"<div disabled=\\"\\" style=\\"--role-color-outline:hsla(0, 0%, 100%, 0.7);--role-color-outline-alt:hsla(0, 0%, 100%, 0.4);--role-color-active:hsl(0, 0%, 100%);--role-color-base:hsl(0, 0%, 100%)\\" title=\\"This role has unsafe permissions.\\" class=\\"rolestyled-r4hjov-0 kfzQxT\\">Test Role</div>"`;

exports[`<Role /> when disabled, handles touch hover events 1`] = `"<div disabled=\\"\\" style=\\"--role-color-outline:hsla(0, 0%, 100%, 0.7);--role-color-outline-alt:hsla(0, 0%, 100%, 0.4);--role-color-active:hsl(0, 0%, 100%);--role-color-base:hsl(0, 0%, 100%)\\" title=\\"This role has unsafe permissions.\\" class=\\"rolestyled-r4hjov-0 kfzQxT\\">unsafe role<div class=\\"tooltip-aykj5z-0 euOEXl\\">This role has unsafe permissions.</div></div>"`;
